### PR TITLE
Add keychain management

### PR DESCRIPTION
**Pull request by cluster**
The goal of this PR is to add the missing functions to manage keychains.
Needed for https://github.com/archethic-foundation/archethic-cli/issues/2 and https://github.com/archethic-foundation/archethic-cli/issues/3

This PR also make the fields of `Keychain` public

### CODE REVIEW

File: account.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+	for i, key := range authorizedPublicKeys {
+		authorizedKeys[i] = AuthorizedKey{
+			publicKey:          key,
+			encryptedSecretKey: EcEncrypt(aesKey, key),
+		}
+	}
+
+	tx := TransactionBuilder{}
+	tx.SetType(KeychainType)
+	keychain.ToDID()
```
Comment (samuelmanzanera) [inline account.go:26]: What is the purpose of this line ? Maybe a miss ;)

File: account.go [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+	if len(accessOwnerships) == 0 {
+		panic("Keychain doesn't exist")
+	}
+	accessSecret := accessOwnerships[0].Secret
+	accessAuthorizedKeys := accessOwnerships[0].AuthorizedPublicKeys
+
+	var accessSecretKey Hex
+	publicKeyHex := strings.ToUpper(hex.EncodeToString(publicKey))
+	for _, authKey := range accessAuthorizedKeys {
+		if strings.ToUpper(string(authKey.PublicKey)) == publicKeyHex {
```
Comment (samuelmanzanera) [inline account.go:67]: Can't you use `bytes.Equal` instead ?

File: account.go [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+	keychainAddress := AesDecrypt(accessSecretBytes, accessKey)
+
+	keychainOwnerships := client.GetTransactionOwnerships(hex.EncodeToString(keychainAddress))
+
+	keychainSecret := keychainOwnerships[0].Secret
+	keychainAuthorizedKeys := keychainOwnerships[0].AuthorizedPublicKeys
+
+	var keychainSecretKey Hex
+	for _, authKey := range keychainAuthorizedKeys {
+		if strings.ToUpper(string(authKey.PublicKey)) == publicKeyHex {
```
Comment (samuelmanzanera) [inline account.go:91]: Same here ?

File: account.go [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+	var accessSecretKey Hex
+	publicKeyHex := strings.ToUpper(hex.EncodeToString(publicKey))
+	for _, authKey := range accessAuthorizedKeys {
+		if strings.ToUpper(string(authKey.PublicKey)) == publicKeyHex {
+			accessSecretKey = authKey.EncryptedSecretKey
+		}
+	}
+
+	accessSecretKeyBytes, err := hex.DecodeString(string(accessSecretKey))
```
Comment (samuelmanzanera) [inline account.go:72]: Why can't you use it as bytes right away ?

File: account.go (hunk 1 above)
Comment (cluster) [reply account.go:26]: eheh well spotted! 😅

File: account.go (hunk 4 above)
Comment (cluster) [reply account.go:72]: good point!

**Review by samuelmanzanera — APPROVED** [review]

**Pull request merged**